crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -174,6 +185,7 @@ impl SoftwareManagerActor {
                     error!("{:?}", err);
                 }
             }
+            SoftwareCommand::SoftwareCommandMetadata(_) => unreachable!(),
```

Comment:
This is unreachable as long as the `tedge_operation_converter` sends no such commands, as of now.

I would consider either to do nothing or to return the actual list of supported software types.

Reply:
I'd simply change to return nothing as it's the simplest.